### PR TITLE
Wrote test for issue 19358: KeyError from dsolve

### LINKED ISSUE #19358: KeyError from dsolve
From SO:
https://stackoverflow.com/questions/61875864/keyerror-expt-using-dsolve-from-sympy-for-simple-ode

This error appears in sympy-1.5.1:
```julia
In [1]: from sympy import symbols, Function, dsolve, exp, Eq 
   ...: a, b, t = symbols('a b t') 
   ...: Y = Function('Y')(t) 
   ...: #f = Function('f')(t) # works 
   ...: #f = exp(a*t) # works 
   ...: f = exp(t-a) # KeyError: exp(t) 
   ...: #f = exp(t-2) # works 
   ...: odeY = Eq( Y.diff(t,t), b*Y.diff(t) + f ) 
   ...: dsolve(odeY,Y)                                                                                                            
---------------------------------------------------------------------------
KeyError
```

It is already fixed in current master but a regression test should be added to make sure that this case continues to be correctly solved in sympy in the future.

This can be closed if a test is added for this:
```julia
In [1]: Y = Function('Y')                                                                                                         

In [2]: a, b, t = symbols('a, b, t')                                                                                              

In [3]: eq = Eq(Derivative(Y(t), (t, 2)), b*Derivative(Y(t), t) + exp(-a + t))                                                    

In [4]: eq                                                                                                                        
Out[4]: 
  2                             
 d            d           -a + t
───(Y(t)) = b⋅──(Y(t)) + ℯ      
  2           dt                
dt                              

In [5]: sol = dsolve(eq)                                                                                                          

In [6]: sol                                                                                                                       
Out[6]: 
                       -a + t
                b⋅t   ℯ      
Y(t) = C₁ + C₂⋅ℯ    - ───────
                       b - 1 

In [7]: checkodesol(eq, sol)                                                                                                      
Out[7]: (True, 0)
```

I'm marking this as easy to fix.

### PR DESCRIPTION
**Pull request by pecan-pine**
#### References to other Issues or PRs

Fixes #19358

#### Brief description of what is fixed or changed
Checks that dsolve doesn't produce a KeyError for a certain differential equation. 

#### Other comments
I tested that this test passes the current Sympy version (as from github), but I am not sure how to test that it fails the old version (1.5.1). I did test that the issue exists in 1.5.1, though.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by czgdp1807** [comment]
I believe this doesn't need a release notes entry as nothing is changed for the end user.

**Comment by pecan-pine** [comment]
The sympy-bot seems to want release notes... Should I ignore the bot and take out the notes?

**Comment by namannimmo10** [comment]
> The sympy-bot seems to want release notes... Should I ignore the bot and take out the notes?

Please write "NO ENTRY".

File: sympy/solvers/ode/tests/test_ode.py [hunk 1]
```diff
@@ -3243,7 +3243,7 @@ def test_issue_9446():
     [Eq(f(x), C1 + pi*x - Integral(asin(f(2*x)), x)), Eq(f(x), C1 + Integral(asin(f(2*x)), x))]
 
     assert integrate(-asin(f(2*x)+pi), x) == -Integral(asin(pi + f(2*x)), x)
-    
+
```
Comment (namannimmo10) [inline sympy/solvers/ode/tests/test_ode.py:3246]: Can you leave another blank line here? Just because there's a pattern in this file.

Comment (pecan-pine) [reply sympy/solvers/ode/tests/test_ode.py:3246]: I think I fixed that

**Comment by namannimmo10** [comment]
Please write "Fixes #19358" under references to other issues or PRs.

**Comment by oscarbenjamin** [comment]
This looks good except that the tests for dsolve are currently being refactored. This ODE is an example that is solved by the nth linear constant coefficients undetermined coefficients solver:
```julia
In [6]: classify_ode(eq)                                                                                                          
Out[6]: 
('nth_linear_constant_coeff_undetermined_coefficients',
 'nth_linear_constant_coeff_variation_of_parameters',
 'nth_order_reducible',
 'nth_linear_constant_coeff_variation_of_parameters_Integral')
```
The other examples for this solver in the refactored tests are here:
https://github.com/sympy/sympy/blob/master/sympy/solvers/ode/tests/test_single.py#L949

This example should be added at the bottom of that function in the refactored form along with the link to the issue in a comment e.g.:
```python
    # https://github.com/sympy/sympy/issues/19358
    'undet_29': {...}
```

**Comment by oscarbenjamin** [comment]
The new test in test_single.py looks good. I think we don't need the old test in test_ode.py now though.

**Comment by pecan-pine** [comment]
I took out the change to test_ode.py in https://github.com/sympy/sympy/pull/19370. Should I delete this pull request then?

**Comment by oscarbenjamin** [comment]
Yes, you can just close this if the work is somewhere else now

**Comment by pecan-pine** [comment]
How do I remove my edit of test_ode.py while keeping the changes for test_single.py?

**Pull request closed without merging**